### PR TITLE
Log Brave headers to console if configured

### PR DESCRIPTION
**Pull request by aekeus**

### CODE REVIEW

File: src/lib/headers.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const _ = require('underscore')
+
+const potentiallyInspectBraveHeaders = (request) => {
+  if (!process.env.INSPECT_BRAVE_HEADERS) return
+  _.each(request.headers, (v, k) => {
+    if (k.match(/^x-brave-/)) {
```
Comment (keur) [inline src/lib/headers.js]: regex is a bit excessive here? how about `k.startsWith("x-brave-")`

File: src/lib/headers.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const _ = require('underscore')
+
+const potentiallyInspectBraveHeaders = (request) => {
+  if (!process.env.INSPECT_BRAVE_HEADERS) return
```
Comment (keur) [inline src/lib/headers.js]: We already have an environment variable for logging all headers here https://github.com/brave/vault-updater/blob/ba3fa630ad5e3b551bc5e4909a8f5d876fc80170/src/index.js#L96-L101

File: src/controllers/ios.js [hunk 3]
```diff
@@ -31,6 +31,7 @@ exports.setup = (runtime) => {
     path: '/1/usage/ios',
     config: {
       handler: function (request, reply) {
+        headers.potentiallyInspectBraveHeaders(request)
```
Comment (keur) [inline src/controllers/ios.js]: Why not add this code [here](https://github.com/brave/vault-updater/blob/ba3fa630ad5e3b551bc5e4909a8f5d876fc80170/src/index.js#L98)? Then we can get it for all paths if necessary You can prefix with `request.route.path` and then filter just by ios in Papertrail.

File: src/lib/headers.js (hunk 2 above)
Comment (keur) [inline src/lib/headers.js]: This is really questionable formatting...

File: src/lib/headers.js (hunk 1 above)
Comment (aekeus) [reply src/lib/headers.js]: Seems reasonable

File: src/lib/headers.js (hunk 2 above)
Comment (aekeus) [reply src/lib/headers.js]: LOG_HEADERS should be deprecated

File: src/controllers/ios.js (hunk 3 above)
Comment (aekeus) [reply src/controllers/ios.js]: Yes, that may be better

File: src/lib/headers.js (hunk 2 above)
Comment (aekeus) [reply src/lib/headers.js]: More details please

File: src/controllers/braveCore.js [hunk 4]
```diff
@@ -51,7 +52,9 @@ exports.setup = (runtime) => {
     path: '/1/usage/brave-core',
     config: {
       handler: (request, reply) => {
-        const usage = buildUsage(request)
+        headers.potentiallyInspectBraveHeaders(request)
```
Comment (keur) [inline src/controllers/braveCore.js]: Adding this to core too really makes me think we should be doing it for every route in `serv.listener.on('request', (request, event, tags)`

File: src/controllers/braveCore.js [hunk 5]
```diff
@@ -51,7 +52,9 @@ exports.setup = (runtime) => {
...
+        let usage = buildUsage(request)
+        usage = headers.potentiallyStoreBraveHeaders(request, usage)
```
Comment (keur) [inline src/controllers/braveCore.js]: This method too could be called for every request.

File: src/lib/headers.js (hunk 2 above)
Comment (keur) [reply src/lib/headers.js]: I would move the `return` to the next line, indent

File: src/controllers/braveCore.js (hunk 5 above)
Comment (aekeus) [reply src/controllers/braveCore.js]: Right now, I only want headers stored for iOS, but I agree in principle.

**Pull request merged**